crates/librqbit/src/torrent_state/live/mod.rs

```diff
@@ -1018,6 +1020,12 @@ impl PeerHandler {
 
         pe.value_mut().state.set(PeerState::Dead, &pstats);
 
+        if self.incoming {
+            // do not retry incoming peers
+            debug!("incoming peer {handle} died, not re-queueing");
```

Comment:
nit: at least for the sake of consistent style please put handle into a parameter, e.g. debug!(peer=handle, "incoming peer died, not re-queueing");